src/Workspaces/CoreTest/ExtensionsTests/DocumentExtensionsTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.ExtensionsTests
+{
+    public class DocumentExtensionsTests
+    {
+        private readonly AdhocWorkspace _ws = new AdhocWorkspace();
+        private readonly Project _emptyProject;
+
+        public DocumentExtensionsTests()
+        {
+            _emptyProject = _ws.AddProject(
```

Comment:
Take a look at the `TestWorkspace` class.  It makes it simple to construct a workspace with the right services, and specify spans/positions within the documents.